src/Mod/TechDraw/App/DrawViewPart.h

```diff
@@ -90,6 +97,10 @@ class TechDrawExport DrawViewPart : public DrawView
     double getBoxX(void) const;
     double getBoxY(void) const;
     virtual QRectF getRect() const;
+    virtual DrawViewSection* getSectionRef() const;                    //is there a ViewSection based on this ViewPart?
+    Base::Vector3d getUDir(void)  {return uDir;}                       //paperspace X

```

Comment:
Thanks for the tips.  Changes will be in the next TechDraw update.

Chris

On Mon, Aug 29, 2016 at 7:43 AM, wwmayer <notifications@github.com> wrote:

> In src/Mod/TechDraw/App/DrawViewPart.h
> <https://github.com/FreeCAD/FreeCAD/pull/266#discussion_r76593287>:
>
> > @@ -90,6 +97,10 @@ class TechDrawExport DrawViewPart : public DrawView
> >      double getBoxX(void) const;
> >      double getBoxY(void) const;
> >      virtual QRectF getRect() const;
> > +    virtual DrawViewSection* getSectionRef() const;                    //is there a ViewSection based on this ViewPart?
> > +    Base::Vector3d getUDir(void)  {return uDir;}                       //paperspace X
>
> uDir, VDir, wDir are member variables and thus you could return a
> const-reference to avoid the creation of a tmp. copy.. Also out of
> const-correctness reasons these methods should be marked as const.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FreeCAD/FreeCAD/pull/266/files/e2945be805b7593c551293def1b23e612f82654a#r76593287>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADa8WbBdrgEwCPoUI5imQjY8KPEkkCzVks5qksXPgaJpZM4Jtl1h>
> .
>